app/models/target.rb

```diff
@@ -1,6 +1,16 @@
 class Target < ApplicationRecord
   belongs_to :topic
-  belongs_to :users
+  belongs_to :user
   validates :title, :radius, :lat, :lng, :topic_id,  presence: true
   validates :radius, numericality: { greater_than: 0 }
+  
+  validate :validate_target_limit, on: :create
+
+  private
+
+  def validate_target_limit
+    if user.targets.count >= 3
+      errors.add(:base, "You have reached the maximum limit of targets.")
```

Comment:
You can move this text to en.yml and use it here, something like:

```suggestion
      errors.add(:user, I18n.t('model.target.errors.max_targets'))
```

See https://guides.rubyonrails.org/i18n.html